tests/end-to-end/cli/filter/filter-dataprovider-by-phpstorm-regex-number-key.phpt

```diff
@@ -0,0 +1,72 @@
+--TEST--
+https://github.com/sebastianbergmann/phpunit/pull/6364
+--FILE--
+<?php declare(strict_types=1);
+
+// Simulate how PHPStorm runs a test class.
+$process = proc_open(
+    [
+        PHP_BINARY,
+        __DIR__ . '/../../../../phpunit',
+        '--do-not-cache-result',
+        '--no-configuration',
+        '--filter',
+        'PHPUnit\\\\TestFixture\\\\DataProviderFilterTest',
+        '--test-suffix',
+        'DataProviderFilterTest.php',
+        __DIR__ . '/../../../_files',
+        '--teamcity',
+    ],
+    [
+        1 => ['pipe', 'w'],
+    ],
+    $pipes,
+);
+
+$stdout = stream_get_contents($pipes[1]);
+fclose($pipes[1]);
+proc_close($process);
+
+if (preg_match("/##teamcity\\[testStarted name='testTrue with data set #1' locationHint='([^']+)'/", $stdout, $matches) !== 1) {
+    echo "Failed to find locationHint.\n";
+    echo $stdout;
+
+    return 0;
+}
+
+if (preg_match('#php_qn://(?:[A-Z]:)?[^:]*::\\\\(.*)#', $matches[1], $locationHintMatches) !== 1) {
+    echo "Failed to parse locationHint.\n";
+    echo $matches[1];
+
+    return 0;
+}
+
```

Comment:
Should this not be two separate PHPT files? It looks wrong to me to invoke PHPUnit twice in a single PHPT file, once using `proc_open()` and once using `run()`.

Reply:
Thank you.